components/automate-deployment/pkg/server/automate_config.go

```diff
@@ -235,3 +254,139 @@ func (s *server) updateUserOverrideConfigFromRestoreBackupRequest(req *api.Resto
 
 	return nil
 }
+
+func setConfigForRedirectLogs(req *api.PatchAutomateConfigRequest, existingCopy *deployment.AutomateConfig) error {
+
+	//Set the config if already not set
+	if req.GetConfig().GetGlobal().GetV1().GetLog().GetRedirectSysLog().GetValue() == true &&
+		existingCopy.GetGlobal().GetV1().GetLog().GetRedirectSysLog().GetValue() == false {
+
+		err := createConfigFileForAutomateSysLog(req.GetConfig().GetGlobal().GetV1().GetLog().GetRedirectLogFilePath().GetValue())
+		if err != nil {
+			return status.Error(codes.Internal, "Failed to create configuration into syslog")
+		}
+		err = restartSyslogService()
+		if err != nil {
+			return status.Error(codes.Internal, "Failed to restart syslog service")
+		}
+
+		if err = runLogrotateConfig(req); err != nil {
+			logrus.Errorf("cannot configure log rotate: %v", err)
+			return err
+		}
+	}
+
+	//Rollback the config if requested
+	if req.GetConfig().GetGlobal().GetV1().GetLog().GetRedirectSysLog().GetValue() == false &&
+		existingCopy.GetGlobal().GetV1().GetLog().GetRedirectSysLog().GetValue() == true {
+		err := removeConfigFileForAutomateSyslog()
+		if err != nil {
+			return status.Error(codes.Internal, "Failed to remove configuration into syslog")
+		}
+		err = restartSyslogService()
+		if err != nil {
+			return status.Error(codes.Internal, "Failed to restart syslog service while removing config file for syslog")
+		}
+
+		if err = rollbackLogrotate(); err != nil {
+			logrus.Errorf("cannot rollback logrotate: %v", err)
+			return err
+		}
+
+	}
+
+	return nil
+}
+
+func restartSyslogService() error {
```

Comment:
please add a description to all the new methods, at least one or two liners describe the actual operation we are doing over that method.